### PR TITLE
HDDS-3627. Remove FilteredClassloader and replace with maven based hadoop2/hadoop3 ozonefs generation

### PR DESCRIPTION
**Pull request by elek**
## What changes were proposed in this pull request?

As described in the [parent issue](https://issues.apache.org/jira/browse/HDDS-3458), the final step is to create a Hadoop independent shaded client and hadoop2/hadoop3 related separated client jars.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3627

## How was this patch tested?

Patch re-enables the hadoop27 mapreduce test which was disabled long time ago. All the hadoop31/hadoop32/hadoop27 mapreduce acceptance tests are passed on my fork.

### CODE REVIEW

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java [hunk 1]
```diff
@@ -145,18 +145,6 @@ public void initialize(URI name, Configuration conf) throws IOException {
...
-      // /ozonefs.txt, otherwise the default is false. It could be overridden.
-      boolean defaultValue =
-          BasicOzoneFileSystem.class.getClassLoader()
-              .getResource("ozonefs.txt")
-              != null;
-
-      //Use string here instead of the constant as constant may not be available
-      //on the classpath of a hadoop 2.7
-      boolean isolatedClassloader =
-          conf.getBoolean("ozone.fs.isolated-classloader", defaultValue);
```
Comment (xiaoyuyao) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:158]: Can you remove OzoneConfigKeys#OZONE_FS_ISOLATED_CLASSLOADER as well?

Comment (elek) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:158]: done.

**Comment by elek** [comment]
Thanks the review @xiaoyuyao 

Yes, all the documentation should be updated. I created a separated issue: https://issues.apache.org/jira/browse/HDDS-3704 (blocker for next release) for documentation. This patch is already big enough. (and merging it can help me to check the integration with the ofs patch).

I think the Spark+k8s part of documentation should be retested

**Comment by xiaoyuyao** [comment]
@elek Agree. If you could check and confirm the CI clean, I think we should merge this one given the size of the patch.

**Comment by elek** [comment]
Thanks the review @xiaoyuyao, I am merging it now.

**Comment by ChenSammi** [comment]
Thanks @elek for providing a clean and elegant hadoop2/hadoop3 client.  I have verified it in our internal hadoop 2.x production cluster.

**Pull request merged**